Lib/asyncio/locks.py

```diff
@@ -105,19 +108,16 @@ async def acquire(self):
         fut = self._get_loop().create_future()
         self._waiters.append(fut)
 
-        # Finally block should be called before the CancelledError
-        # handling as we don't want CancelledError to call
-        # _wake_up_first() and attempt to wake up itself.
         try:
-            try:
-                await fut
-            finally:
-                self._waiters.remove(fut)
-        except exceptions.CancelledError:
+            await fut
+        except BaseException:
```

Comment:
Fair point.  I quite understand that you are reluctant to touch this just to humour an eccentric experimental library which is trying to push the envelope of the original design.   And I'm happy to have this rejected as long as we have had the chance to discuss it.  I'll add a comment here in the mean time, just for safety.

Reply:
Sure.  Of course, IMO, it would be even greater if in some future version, we would have something like:
```python
class InterruptError(BaseException):
    pass

class CancelledError(InterruptError):
   pass
```

but that is future music :)